From 420351fbdea84f7527cc96d714bc2a1ae3673274 Mon Sep 17 00:00:00 2001
From: tpaviot <tpaviot@gmail.com>
Date: Sun, 3 Jun 2012 16:50:48 +0200
Subject: [PATCH] [clang-warning-fix][dynamic-class-memaccess]

Destination for this 'memcpy' call is a pointer to a dynamic class;
vtable pointer will be overwritten. Fix: explicitely cast the pointer
to silence this warning.
---
 inc/NCollection_Vector.hxx |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

Index: oce/inc/NCollection_Vector.hxx
===================================================================
--- oce.orig/inc/NCollection_Vector.hxx
+++ oce/inc/NCollection_Vector.hxx
@@ -293,7 +293,7 @@ template <class TheItemType> class NColl
       (MemBlock *) aSelf.myAllocator->Allocate(aCapacity * sizeof(MemBlock));
     Standard_Integer i = 0;
     if (aSource != NULL) {
-      memcpy (aData, aSource, aSize * sizeof(MemBlock));
+      memcpy ((void*)aData, aSource, aSize * sizeof(MemBlock));
       i = aSize;
     }
     while (i < aCapacity)
